### PR TITLE
3152 sol added

### PR DESCRIPTION
**Pull request by AmruthaPariprolu**
## Related Issue

[Cite any related issue(s) this pull request addresses. If none, simply state "None”]

## Description

[Please include a brief description of the changes or features added]

## Type of PR

- [ ] Bug fix
- [x] Feature enhancement
- [ ] Documentation update
- [ ] Security enhancement
- [ ] Other (specify): _______________

## Screenshots / Videos (if applicable)

[Attach any relevant screenshots or videos demonstrating the changes]

## Checklist
- [x] I have performed a self-review of my code.
- [x] I have read and followed the Contribution Guidelines.
- [x] I have tested the changes thoroughly before submitting this pull request.
- [x] I have provided relevant issue numbers, screenshots, and videos after making the changes.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have followed the code style guidelines of this project.
- [x] I have checked for any existing open issues that my pull request may address.
- [x] I have ensured that my changes do not break any existing functionality.
- [x] Each contributor is allowed to create a maximum of 4 issues per day. This helps us manage and address issues efficiently.
- [x] I have read the resources for guidance listed below.
- [x] I have followed security best practices in my code changes.

## Additional Context

[Include any additional information or context that might be helpful for reviewers.]

## Resources for Guidance

Please read the following resources before submitting your contribution:

- [ ] [Code Harbor Hub Community Features](https://www.codeharborhub.live/community/features)
- [ ] [Docusaurus Documentation](https://docusaurus.io/docs/create-doc)
- [ ] [React.js Documentation](https://legacy.reactjs.org/docs/getting-started.html)
- [ ] [Markdown Guide](https://www.markdownguide.org/)
- [ ] [MDX Documentation](https://mdxjs.com/docs/)
- [ ] [Mermaid Documentation](https://mermaid.js.org/)

**Resources for DSA Features on Our Site:**

- [ ] [Tabs: Switch Languages](https://www.codeharborhub.live/community/features#multi-language-support-code-blocks)
- [ ] [Line Numbering](https://www.codeharborhub.live/community/features#line-numbering)
- [ ] [Math Equations](https://www.codeharborhub.live/community/features#math-equations-with-katex) 

## Contribution Guidelines

Thank you for considering contributing to our project! To ensure smooth collaboration and effective contribution management, please adhere to the following guidelines:

### Issue Creation

1. **Limit on Issues:**
   - Each contributor is allowed to create a maximum of **4 issues per day**. This helps us manage and address issues efficiently.

### Contribution Levels

2. **Basic Contributions:**
   - This project is primarily focused on documentation. Most of the setup has been completed, so contributors will generally need to work on basic code tasks, such as writing tests.
   - For these tasks, issues will be assigned the **level1** label.

3. **Acknowledging Hard Work:**
   - If a contributor puts in significant effort on a task, the issue will be upgraded to **level2**. This is our way of recognizing and appreciating extra effort.

4. **Feature Additions and Component Work:**
   - Contributors working on new features or components using JSX/TSX will be assigned a level based on the complexity and quality of their work.
   - The more complex and valuable the contribution, the higher the level assigned.

### Level Definitions

- **Level1:**
  - Tasks are straightforward, such as fixing minor bugs, writing tests, or making simple documentation updates.
- **Level2:**
  - Tasks require more effort, such as addressing complex bugs, improving existing features, or making substantial documentation improvements.
- **Level3:**
  - Tasks are highly complex and involve significant new feature development, major refactoring, or extensive contributions to the project’s core components.

We look forward to your contributions and appreciate your effort in helping us improve the project!

### CODE REVIEW

File: dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+Explanation:
+
+The subarray is [4,3,1]. 3 and 1 are both odd. So the answer to this query is false.
+The subarray is [1,6]. There is only one pair: (1,6) and it contains numbers with different parity. So the answer to this query is true.
+```
+
+### Constraints
+- 1 <= nums.length <= 105
+ - 1 <= nums[i] <= 105
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md:54]: replace `1 <= nums[i] <= 105` to ``` `1 <= nums[i] <= 105` ```

File: dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+- 1 <= queries.length <= 105
+- queries[i].length == 2
+- 0 <= queries[i][0] <= queries[i][1] <= nums.length - 1
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md:57]: replace `0 <= queries[i][0] <= queries[i][1] <= nums.length - 1` to ``` `0 <= queries[i][0] <= queries[i][1] <= nums.length - 1` ```

File: dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        result.append(is_special)
+    return result
+
+# Example usage
+nums = [3, 4, 1, 2, 6]
+queries = [[0, 4]]
+print(check_special_subarray(nums, queries))  # Output: [False]
+
+```
+
```
Comment (Ajay-Dhangar) [inline dsa-solutions/lc-solutions/3100-3199/3152- Special-Array-II.md:152]: ### Bade practices

```md

# Complexity Analysis
## Time Complexity: $O(q*n)$ where q is no of queries and n is array length.
### Reason: iterating throught the array takes $O(n)$ we do it 'q' times so $O(q*n)$
# Space Complexity: $O(q)$
### Reason: space required to store array of size q.
```

### Good Practices

```md
## Complexity Analysis
- **Time Complexity:** $O(q*n)$ where q is no of queries and n is array length.
   > **Reason:** iterating throught the array takes $O(n)$ we do it 'q' times so $O(q*n)$
- **Space Complexity:** $O(q)$
   > **Reason:** space required to store array of size q.
```

**Review by Ajay-Dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

**Comment by AmruthaPariprolu** [comment]
made required changes pls verify

**Pull request merged**